.circleci/config.yml

```diff
@@ -30,6 +30,10 @@ defaults: &defaults
         name: Stack upgrade
         command: stack upgrade
 
+    - run:
+        name: Cabal update
+        command: cabal v2-update
```

Comment:
For repeatability, we might consider something like

```
cabal v2-update 'hackage.haskell.org,2019-12-26T09:49:12Z'
```

Reply:
this is just an experiment if this fixes the errors.
The tests are mainly executed with stack in mind, e.g. generated `hie.yaml` all use stack. So, actually, any cabal execution in this CI build is wrong, imo.
To fix this, we have to tackle #1431.

Reply:
Found the problems and removed that line again. Only one small test-failure.